awxkit/awxkit/api/pages/api.py

```diff
@@ -100,7 +100,13 @@ def _export(self, _page, post_fields):
             else:
                 if post_fields[key]['type'] == 'id' and _page.json.get(key) is not None:
                     log.warning("Related link %r missing from %s, attempting to reconstruct endpoint.", key, _page.endpoint)
-                    resource = getattr(self, key, None)
+                    res_pattern, resource = getattr(resources, key, None), None
+                    if res_pattern:
+                        try:
+                            top_level = res_pattern.split('/')[3]
+                            resource = getattr(self, top_level, None)
```

Comment:
This code is kind of hard to read.

The best I can reconstruct, the reason for this is to handle

```python
>>> from awxkit.api.resources import resources
>>> resources.application
'/api/v2/applications/\\d+/'
>>> resources.application.split('/')[3]
'applications'
```

Is that right? This is to add an "s" to "application"?

Reply:
To 'organization', yes.

Reply:
I didn't want to assume that I could just always concatenate an 's' and get the correct endpoint, e.g. 'inventory' -> 'inventories'.